gtsam/sfm/PowerMethod.h

```diff
@@ -0,0 +1,367 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010-2019, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file   PowerMethod.h
+ * @date   Sept 2020
+ * @author Jing Wu
+ * @brief  accelerated power method for fast eigenvalue and eigenvector computation
+ */
+
+#pragma once
+
+#include <gtsam/base/Matrix.h>
+#include <gtsam/base/Vector.h>
+#include <gtsam/dllexport.h>
+
+#include <Eigen/Core>
+#include <Eigen/Sparse>
+#include <map>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+#include <algorithm>
+#include <cmath>
+#include <chrono>
+#include <random>
+
+namespace gtsam {
+
+using Sparse = Eigen::SparseMatrix<double>;
+
+/* ************************************************************************* */
+/// MINIMUM EIGENVALUE COMPUTATIONS
+
+struct PowerFunctor {
```

Comment:
This strikes me as unnecessarily complicated ... is a struct _really_ necessary here?  Wouldn't it be much simpler to have a single (standalone) function  with a signature like:

``` bool powerMethod(const SparseMatrix& A, const Vector& v0, int max_iters, double eps, Vector& v, int& num_iters, bool accelerated = false, double beta = 0) ```

Reply:
It's necessary as we will get rid of big sparse matrices alltogether

Reply:
Ahhh right.  In that case (and just as a suggestion): might it still make more sense to separate whatever functor you want to use for matrix-vector multiplies from the logic of the power method?  (I say this because I could see both the graphical matrix-vector multiply functor *and* the power method being independently useful for many things -- might be easier from the perspective of modularity and code reuse not to tie their implementations together in a *single* struct ... )